src/main/java/io/github/thebusybiscuit/slimefun4/utils/tags/SlimefunTag.java

```diff
@@ -0,0 +1,325 @@
+package io.github.thebusybiscuit.slimefun4.utils.tags;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Tag;
+import org.bukkit.block.data.Waterlogged;
+
+import io.github.thebusybiscuit.slimefun4.api.exceptions.TagMisconfigurationException;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.blocks.BlockPlacer;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines.CropGrowthAccelerator;
+import io.github.thebusybiscuit.slimefun4.implementation.items.multiblocks.miner.IndustrialMiner;
+import io.github.thebusybiscuit.slimefun4.implementation.items.tools.ClimbingPick;
+import io.github.thebusybiscuit.slimefun4.implementation.items.tools.ExplosiveShovel;
+import io.github.thebusybiscuit.slimefun4.implementation.items.tools.PickaxeOfTheSeeker;
+import io.github.thebusybiscuit.slimefun4.implementation.items.tools.PickaxeOfVeinMining;
+import io.github.thebusybiscuit.slimefun4.implementation.items.tools.SmeltersPickaxe;
+
+/**
+ * This enum contains various implementations of the {@link Tag} interface.
+ * Most of them serve some purpose within Slimefun's implementation, some are just pure
+ * extensions of the default Minecraft tags.
+ * The actual tag files are located in the {@literal /src/main/resources/tags} directory
+ * and follow Minecraft's tags.json format.
+ * 
+ * @author TheBusyBiscuit
+ * 
+ * @see TagParser
```

Comment:
```suggestion
 * @see TagParser
 * @see Material
```

Reply:
That doesn't really make much sense here, we generally use `@see` to link relevant classes in the same project.
As the main intent behind that parameter is to give additional context to the code to better understand how it plays into the bigger context of the project.

Reply:
> That doesn't really make much sense here, we generally use `@see` to link relevant classes in the same project.
> As the main intent behind that parameter is to give additional context to the code to better understand how it plays into the bigger context of the project.

Exactly. SlimefunTag's center is Material's anyway.

Reply:
Material is from Bukkit though, not from Slimefun,...